plugins/package-managers/maven/src/main/kotlin/Tycho.kt

```diff
@@ -425,20 +428,54 @@ private fun createPackageFromLocalArtifact(artifact: Artifact, localRepositoryRo
 }
 
 /**
- * Parse this string with a source references manifest header to a [Scm] structure.
+ * An enumeration that defines the properties to be extracted from a source reference manifest header.
+ */
+private enum class ScmProperties {
+    /** The connection URL. */
+    CONNECTION,
+
+    /** The tag in the source code management system. */
+    TAG,
+
+    /** The path of this component in a monorepo. */
+    PATH;
+
+    companion object {
+        /** A mapping from string-based keys to the constants of this enumeration. */
+        private val propertiesByKey = entries.associateBy { it.name }
+
+        /**
+         * Return the [ScmProperties] for the given [key], or null if no such property exists.
+         */
+        fun fromKey(key: String): ScmProperties? = propertiesByKey[key.uppercase()]
```

Comment:
Why not just use `valueOf()` instead of introducing this function?

Reply:
`valueOf()` throws an exception for unknown keys, but such keys should just be ignored. I have removed the companion object from the `enum` and handle unknown keys during parsing (by wrapping access in a `runCatching` block).